Add storage subsection for admin tasks chapter

sergiu121 opened the pull request.

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -2,29 +2,135 @@ Gestionarea spațiului de stocare partajat
 Stocarea partajată folosind Dropbox
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Dropbox este o soluție care se folosește de un server în Internet care stochează fișierele noastre, ca apoi acestea să fie replicate pe fiecare calculator client.
```
fără noastre

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -2,29 +2,135 @@ Gestionarea spațiului de stocare partajat
 Dropbox este o soluție care se folosește de un server în Internet care stochează fișierele noastre, ca apoi acestea să fie replicate pe fiecare calculator client.
 
-Îi vom pune să instaleze Dropbox pentru a sincroniza fișierele de pe workstation și mașina locală.
+Este necesar să ne creăm un cont pentru a folosi serviciul Dropbox și să îl activăm.
```
Este necesară crearea și activarea unui cont pentru ...

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -2,29 +2,135 @@ Gestionarea spațiului de stocare partajat
-Îi vom pune să instaleze Dropbox pentru a sincroniza fișierele de pe workstation și mașina locală.
+Este necesar să ne creăm un cont pentru a folosi serviciul Dropbox și să îl activăm.
+
+Dropbox oferă o aplicație care rulează în linie de comandă pe care o vom descărca pentru a sincroniza sistemul de fișiere de pe serverele Dropbox într-un director local.
```
Unde o descărcăm? local, remote, să ai în vedere să precizezi asta

Sergiu121 replied:
Se vede din promp că folosim ``uso``, nu?

razvand replied:
Nu e suficient. Spune și în text. Fă viața ușoară cititorului.

Sergiu121 replied:
Oricum, am pus clarificare mai sus.

razvand reviewed (changes requested):
Îmi place. Am lăsat observații.

O observație care se aplică și la celelalte secțiuni: Găsește o denumire pentru stația locală și pentru stația remote (`local` / `remote` sau altceva) și definește-o de la bun început în capitol. Apoi folosește-o consecvent pe parcursul capitolului. Că acum folosești "stația curentă", "sistemul local", "stația de la distanță", "workstationul", "laptopul" și cauzează o confuzie.

Și când apare că se rulează o comandă sau se execută o acțiune, să fie clar unde se rulează: pe ``local`` sau ``remote``.

Sergiu121 commented on chapters/task-admin/storage.rst:
```diff
@@ -2,29 +2,135 @@ Gestionarea spațiului de stocare partajat
-O altă nevoie pe care o avem, dacă suntem programatori este editarea codului la distanță, deoarece majoritatea programatorilor folosesc IDE-uri în mediu grafic, care nu pot rula mereu eficient de la distanță.
+Cu toate că noi vom rula aplicații pe serverul de la distanță, noi avem nevoie de acces la spațiul de stocare al acestuia, deoarece vrem ca într-un final să urmărim rezultatul procesării și eventual să îl analizăm folosind diferite utilitate grafice.
+O altă nevoie pe care o avem este editarea codului la distanță, deoarece majoritatea programatorilor folosesc IDE-uri în mediu grafic, care nu pot rula mereu eficient de la distanță.
 Soluția la această nevoie este să partajăm spațiul de stocare între serverul de la distanță și laptopul sau stația locală de pe care lucrăm.
```
Nu am în codul sursă, e ceva afișat prost de la GitHub.

Sergiu121 commented on chapters/task-admin/storage.rst:
```diff
@@ -2,29 +2,135 @@ Gestionarea spațiului de stocare partajat
+    root@local:~# echo root@10.10.10.2:/  /mnt  fuse.sshfs  defaults  0  0 > /etc/fstab
+    root@local:~# mount -a
+
+Am scris în fișierul ``/etc/fstab`` folosind comanda ``echo``, iar pentru a monta sistemul de fișiere am folosit comanda ``mount`` cu opțiunea ``-a`` pentru montarea sistemelor de fișiere descrise în fișierul ``/etc/fstab``.
```
Nu vreau să folosesc ``mount``. Am pus ``df``, ca să nu mai introduc alte comenzi noi.

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+    Pentru a parcurge această secțiune este recomandat să descărcați ultima
+    versiune a respository-ului laboratorului. Pentru a descărca ultima versiune
+    a repository-ului rulați comanda ``git pull`` în directorul
+    ``~/uso-lab/labs/10-tasks-admin/lab-containers/``.
```
O propoziție pe rând

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+    a repository-ului rulați comanda ``git pull`` în directorul
+    ``~/uso-lab/labs/10-tasks-admin/lab-containers/``.
+
+    Infrastructura laboratorului este bazată pe containere docker ale căror
```
O propoziție pe rând

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+    Docker Engine pe sistem, scriptul
+    ``~/uso-lab/labs/10-tasks-admin/lab-containers/lab_prepare.sh`` vă va instala aplicația.
+
+    După ce ați terminat de lucrat vă recomandăm să opriți containerele rulând
```
O propoziție pe rând

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
-Soluția la această nevoie este să partajăm spațiul de stocare între serverul de la distanță și laptopul sau stația locală de pe care lucrăm.
+Cu toate că vom rula aplicații pe serverul de la distanță, avem nevoie de acces la spațiul de stocare al acestuia, deoarece vrem ca, într-un final, să urmărim rezultatul procesării și, eventual, să îl analizăm folosind utilitare grafice dedicate.
+O altă nevoie pe care o avem este editarea codului la distanță, deoarece majoritatea programatorilor folosesc IDE-uri în mediu grafic, care nu pot rula mereu eficient de la distanță.
+Soluția la această nevoie este să partajăm spațiul de stocare între serverul ``remote`` și ``local`` de pe care lucrăm.
```
și stația ``local``

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+
+Deoarece nu ne dorim să rulăm comanda ``sshfs`` atunci când vrem să folosim un sistem de fișiere la distanță, vrem să montăm persistent sistemul de fișiere de la distanță, astfel încât această montare să persiste după oprirea stației locale.
+
+Ca să montăm persistent sistemul de fișiere avem nevoie să copiem cheia SSH pe stația de la distanța, deoarece montarea se va face în mod neinteractiv, deci nu vom avea posibilitatea de a introduce parola. <TODO REF NETWORKING>
```
distanță

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+
+Ca să montăm persistent sistemul de fișiere avem nevoie să copiem cheia SSH pe stația de la distanța, deoarece montarea se va face în mod neinteractiv, deci nu vom avea posibilitatea de a introduce parola. <TODO REF NETWORKING>
+
+Pentru a monta persistent sistemul de fișiere vom scrie o intrare în fișierul ``/etc/fstab`` care va conține detalii despre sistemul de fișiere pe care vrem să îl montăm.
```
, vom

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
 -------------------------------------------
 
-SSHFS nu este o soluție bună pentru a face backup fișierelor, deoarece existând o singură replică, ai șters un fișier și au dispărut toate replicile.Pe lângă asta, dacă ai Internet slab, ai acces greu la fișiere (le vrei și local). Și, pe lângă asta, trebuie să ai SSH configurat, care poate necesita tunel etc.
+SSHFS nu este o soluție bună pentru a face backup fișierelor deorece, existând o singură replică, ștergerea locală a unui fișier duce la ștergerea sa și de pe stația remote și, astfel, la pierderea sa.
```
deoarece

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
-O alternativă pentru acest serviciu sunt soluții cum ar fi GoogleDrive, Dropbox, ownCloud sau OneDrive, care for stoca o replică a fișierului pe toate calculatoarele autentificate de pe un anumit cont.
-Avantajul aici este că aceste sisteme oferă suport pentru controlul versiunilor pentru a șterge modificarea anterioară. Cu dezavantajul că trebuie configurate. Și cu dezavantajul că acum informația este duplicată: dublu spațiu ocupat și pot apărea conflicte la modificări.
+O alternativă la folosirea SSHFS sunt soluții cum ar fi GoogleDrive, Dropbox, ownCloud sau One Drive.
+Aceste soluții sochează o replică a fișierului pe toate calculatoarele autentificate de pe un anumit cont.
```
stochează

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,183 @@
+
+Pentru a instala ultimele componente ale clientului Dropbox este necesar să rulăm comanda ``dropbox start -i``.
+După ce am ponit aplicația, este necesar să înregistrăm stația online la contul nostru.
+Vom face asta accesând linkul afișat de aplicație.
```
asta -> acest lucru

razvand reviewed (changes requested):
Am adăugat recenzie.

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+.. note::
+
+    Pentru a parcurge această secțiune este recomandat să descărcați ultima versiune a respository-ului laboratorului.
+    Pentru a descărca ultima versiune a repository-ului rulați comanda ``git pull`` în directorul ``~/uso-lab/labs/10-tasks-admin/lab-containers/``.
```
, rulați

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+    Pentru a descărca ultima versiune a repository-ului rulați comanda ``git pull`` în directorul ``~/uso-lab/labs/10-tasks-admin/lab-containers/``.
+
+    Infrastructura laboratorului este bazată pe containere Docker ale căror imagini vor fi generate pe propriul calculator.
+    Dacă nu veți deja instalat Docker Engine pe sistem, scriptul ``~/uso-lab/labs/10-tasks-admin/lab-containers/lab_prepare.sh`` vă va instala aplicația.
```
veți -> aveți

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+
+    Sintaxa comenzii ``sshfs`` se aseamănă cu sintaxa comenzii ``scp``.
+
+Acest mod te montare a sistemului de fișier este temporară.
```
te -> de

fișier -> fișiere

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+    Sintaxa comenzii ``sshfs`` se aseamănă cu sintaxa comenzii ``scp``.
+
+Acest mod te montare a sistemului de fișier este temporară.
+Atunci când vom opri stația sistemul de fișiere va fi demontat.
```
stația,

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+
+Ca să montăm persistent sistemul de fișiere avem nevoie să copiem cheia SSH pe stația de la distanță, deoarece montarea se va face în mod neinteractiv, deci nu vom avea posibilitatea de a introduce parola. <TODO REF NETWORKING>
+
+Pentru a monta persistent sistemul de fișiere, vom scrie o intrare în fișierul ``/etc/fstab`` care va conține detalii despre sistemul de fișiere pe care vrem să îl montăm.
```
fstab, fișier care va ...

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+    shm                 64M     0   64M   0% /dev/shm
+    root@10.10.10.2:/   16G   14G  539M  97% /mnt
+
+Am scris în fișierul ``/etc/fstab`` folosind comanda ``echo``, iar pentru a monta sistemul de fișiere am folosit comanda ``mount`` cu opțiunea ``-a`` pentru montarea sistemelor de fișiere descrise în fișierul ``/etc/fstab``.
```
Pune un warnind de tipul "kids, do not try this at home. Folosiți editare interactivă, că altfel vă suprascrieți fișierul ``/etc/fstab``, pierdeți ce aveați deja acolo and all hell breaks loose."

Sergiu121 replied:
Nici nu am văzut că am lăsat făra apendare. Am modificat și am zis și de editor.

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+O alternativă la folosirea SSHFS sunt soluții cum ar fi GoogleDrive, Dropbox, ownCloud sau One Drive.
+Aceste soluții stochează o replică a fișierului pe toate calculatoarele autentificate de pe un anumit cont.
+Un alt avantaj al acestora este că oferă suport pentru controlul versiunilor pentru a șterge modificarea anterioară.
+Cu dezavantajul că trebuie configurate și riscul apariției conflictelor la modificări simultane pe noduri diferite.
```
și cu riscul

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+
+Dropbox oferă o aplicație care rulează în linie de comandă pe care o vom descărca pentru a sincroniza sistemul de fișiere de pe serverele Dropbox într-un director local.
+
+Pentru a descărca aplicația Dropbox, pe care o vom descărca folosind comanda ``wget`` și o vom instala folosind comanda ``dpkg`` împreună cu parametrul.
```
Ratezi predicatul principal din fraza de mai sus.

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+.. note::
+
+    Pentru rezolvarea acestui exercițiu, rulați în directorul ``~/uso.git/labs/03-user/lab-containers/`` comanda ``./lab_prepare.sh install dropbox``.
+    Pentru a ne conecta la infrastructura pentru această secțiune vom folosi comanda ``./lab_prepare.sh connect dropbox``.
```
, vom folosi

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
 
-Îi vom pune să instaleze Dropbox pentru a sincroniza fișierele de pe workstation și mașina locală.
+#) Conectați-vă la stația ``dropbox`` și porniți aplicația Dropbox pe aceasta.
+#) Scrieți un fișier numit ``hello.txt`` mesajul ``Hello from remote`` de pe stația ``dropbox``.
```
Cum adică "de pe stația ``dropbox``"?

Sergiu121 replied:
Am clarificat.

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
-Îi vom pune să instaleze Dropbox pentru a sincroniza fișierele de pe workstation și mașina locală.
+#) Conectați-vă la stația ``dropbox`` și porniți aplicația Dropbox pe aceasta.
+#) Scrieți un fișier numit ``hello.txt`` mesajul ``Hello from remote`` de pe stația ``dropbox``.
+Verificați conținutul fișierului de pe stația ``uso``.
```
De ce "de pe" și nu "pe"

razvand commented on chapters/task-admin/storage.rst:
```diff
@@ -1,30 +1,177 @@
+#) Scrieți un fișier numit ``hello.txt`` mesajul ``Hello from remote`` de pe stația ``dropbox``.
+Verificați conținutul fișierului de pe stația ``uso``.
 
 Extra: Stocarea partajată folosind un server privat
```
TODO aici

razvand reviewed (changes requested):
Am adăugat o nouă recenzie.

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge

razvand committed:
task-admin: Add storage section

Add section detailing how to share storage space between local system
and remote workstation.

Signed-off-by: Sergiu Weisz <sergiu121@gmail.com>

Sergiu121 commented:
Published at http://sergiu121.github.io/refs/pull/37/merge
